Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp

```diff
@@ -853,34 +852,32 @@ namespace AzToolsFramework
         {
             return false;
         }
-        
-        // Only allow reparenting the selected entities if they are all under the same instance.
-        // We check the parent entity separately because it may be a container entity and
-        // container entities consider their owning instance to be the parent instance
-        auto prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get();
-        AZ_Assert(prefabPublicInterface, "EntityOutlinerListModel requires a PrefabPublicInterface instance on Initialize.");
-        if (!prefabPublicInterface->EntitiesBelongToSameInstance(selectedEntityIds))
-        {
-            return false;
-        }
 
-        // Disable parenting to a different owning instance
-        auto instanceEntityMapperInterface = AZ::Interface<AzToolsFramework::Prefab::InstanceEntityMapperInterface>::Get();
-        if (instanceEntityMapperInterface)
+        if (auto prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get())
```

Comment:
shouldn't this be
```
if (auto prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get(); prefabPublicInterface != nullptr)
```
?
If it results in a nullptr, the current code will still enter the if loop.

Reply:
While your suggestion will work, original code is also correct. When initializing a variable in an if condition, the if condition's result will be evaluated by converting the value of the variable to bool.

So in this case
```
if (auto prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get())
{
    // A
}
else
{
    // B
}
```
Since `prefabPublicInterface` is a raw pointer, he `B` branch will be executed if `prefabPublicInterface` is initialized to `nullptr`, while `A` will be executed if it has any other value.

Keep in mind that this check is only executed if no semicolon is added to the end. So something like
```
if (auto prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get(); prefabPublicInterface->IsLevelInstanceContainerEntity(entityId))
```
will crash if `prefabPublicInterface` is `nullptr`. So in this case you need to be more verbose and add the extra check
```
if (auto prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get(); prefabPublicInterface && prefabPublicInterface->IsLevelInstanceContainerEntity(entityId))
```